In [ ]:
import pandas as pd
import numpy as np
import matplotlib.pyplot
import plotly.express as px
import plotly.graph_objects as go
#import plotly.io as pio
#pio.templates.default = "plotly_white"

In [ ]:
data = pd.read_csv("/content/drive/MyDrive/apple_products.csv")

In [ ]:
#Read Data
data.head()

Product Name  \
0         APPLE iPhone 8 Plus (Gold, 64 GB)   
1  APPLE iPhone 8 Plus (Space Grey, 256 GB)   
2      APPLE iPhone 8 Plus (Silver, 256 GB)   
3           APPLE iPhone 8 (Silver, 256 GB)   
4             APPLE iPhone 8 (Gold, 256 GB)   

                                         Product URL  Brand  Sale Price  \
0  https://www.flipkart.com/apple-iphone-8-plus-g...  Apple       49900   
1  https://www.flipkart.com/apple-iphone-8-plus-s...  Apple       84900   
2  https://www.flipkart.com/apple-iphone-8-plus-s...  Apple       84900   
3  https://www.flipkart.com/apple-iphone-8-silver...  Apple       77000   
4  https://www.flipkart.com/apple-iphone-8-gold-2...  Apple       77000   

     Mrp  Discount Percentage  Number Of Ratings  Number Of Reviews  \
0  49900                    0               3431                356   
1  84900                    0               3431                356   
2  84900                    0               3431                356   
3  77000                    0              11202                794   
4  77000                    0              11202                794   

                Upc  Star Rating   Ram  
0  MOBEXRGV7EHHTGUH          4.6  2 GB  
1  MOBEXRGVAC6TJT4F          4.6  2 GB  
2  MOBEXRGVGETABXWZ          4.6  2 GB  
3  MOBEXRGVMZWUHCBA          4.5  2 GB  
4  MOBEXRGVPK7PFEJZ          4.5  2 GB

# Data Cleaning

**Check Null Values**

In [ ]:
print("Null Values")
data.isnull().sum()

Null Values


Product Name           0
Product URL            0
Brand                  0
Sale Price             0
Mrp                    0
Discount Percentage    0
Number Of Ratings      0
Number Of Reviews      0
Upc                    0
Star Rating            0
Ram                    0
dtype: int64

The dataset does not have any missing values.

In [ ]:
data.describe()

Sale Price            Mrp  Discount Percentage  Number Of Ratings  \
count      62.000000      62.000000            62.000000          62.000000   
mean    80073.887097   88058.064516             9.951613       22420.403226   
std     34310.446132   34728.825597             7.608079       33768.589550   
min     29999.000000   39900.000000             0.000000         542.000000   
25%     49900.000000   54900.000000             6.000000         740.000000   
50%     75900.000000   79900.000000            10.000000        2101.000000   
75%    117100.000000  120950.000000            14.000000       43470.000000   
max    140900.000000  149900.000000            29.000000       95909.000000   

       Number Of Reviews  Star Rating  
count          62.000000    62.000000  
mean         1861.677419     4.575806  
std          2855.883830     0.059190  
min            42.000000     4.500000  
25%            64.000000     4.500000  
50%           180.000000     4.600000  
75%          3331.000000     4.600000  
max          8161.000000     4.700000

**Min sales** = 29999

**Max sales** = 140900

**Min Mrp** = 39900

**Max Mrp** = 149900

**Min star rating** = 4.5

**Min star rating** = 4.7

# Top 10 highest-rated iPhone on Flipkart in India?

In [ ]:
Highest_rated = data.sort_values(by=['Star Rating'], ascending=False)
Highest_rated = Highest_rated.head(10)

In [ ]:
print(Highest_rated['Product Name'])

20     APPLE iPhone 11 Pro Max (Midnight Green, 64 GB)
17         APPLE iPhone 11 Pro Max (Space Grey, 64 GB)
16    APPLE iPhone 11 Pro Max (Midnight Green, 256 GB)
15               APPLE iPhone 11 Pro Max (Gold, 64 GB)
14              APPLE iPhone 11 Pro Max (Gold, 256 GB)
0                    APPLE iPhone 8 Plus (Gold, 64 GB)
29                     APPLE iPhone 12 (White, 128 GB)
32          APPLE iPhone 12 Pro Max (Graphite, 128 GB)
35                     APPLE iPhone 12 (Black, 128 GB)
36                      APPLE iPhone 12 (Blue, 128 GB)
Name: Product Name, dtype: object


In [ ]:
print(Highest_rated)

                                        Product Name  \
20   APPLE iPhone 11 Pro Max (Midnight Green, 64 GB)   
17       APPLE iPhone 11 Pro Max (Space Grey, 64 GB)   
16  APPLE iPhone 11 Pro Max (Midnight Green, 256 GB)   
15             APPLE iPhone 11 Pro Max (Gold, 64 GB)   
14            APPLE iPhone 11 Pro Max (Gold, 256 GB)   
0                  APPLE iPhone 8 Plus (Gold, 64 GB)   
29                   APPLE iPhone 12 (White, 128 GB)   
32        APPLE iPhone 12 Pro Max (Graphite, 128 GB)   
35                   APPLE iPhone 12 (Black, 128 GB)   
36                    APPLE iPhone 12 (Blue, 128 GB)   

                                          Product URL  Brand  Sale Price  \
20  https://www.flipkart.com/apple-iphone-11-pro-m...  Apple      117100   
17  https://www.flipkart.com/apple-iphone-11-pro-m...  Apple      117100   
16  https://www.flipkart.com/apple-iphone-11-pro-m...  Apple      131900   
15  https://www.flipkart.com/apple-iphone-11-pro-m...  Apple      117100   
14 

# How many rating do the highest-rated iPhone on Flipkart have?

In [ ]:
iPhone = Highest_rated['Product Name'].value_counts()
print(iPhone)


Product Name
APPLE iPhone 11 Pro Max (Midnight Green, 64 GB)     1
APPLE iPhone 11 Pro Max (Space Grey, 64 GB)         1
APPLE iPhone 11 Pro Max (Midnight Green, 256 GB)    1
APPLE iPhone 11 Pro Max (Gold, 64 GB)               1
APPLE iPhone 11 Pro Max (Gold, 256 GB)              1
APPLE iPhone 8 Plus (Gold, 64 GB)                   1
APPLE iPhone 12 (White, 128 GB)                     1
APPLE iPhone 12 Pro Max (Graphite, 128 GB)          1
APPLE iPhone 12 (Black, 128 GB)                     1
APPLE iPhone 12 (Blue, 128 GB)                      1
Name: count, dtype: int64


In [ ]:
label = iPhone.index
counts = Highest_rated['Number Of Ratings']
figure = px.bar(Highest_rated, x=label, y = counts,title = "Number of Ratings Highest Rated iPhone")
figure.show()

# Which iPhone has the highest number of reviews on Flipkart?

In [ ]:
label = iPhone.index
counts = Highest_rated['Number Of Reviews']
figure = px.bar(Highest_rated, x=label, y = counts,title = "Number Of Reviews Highest Rated iPhone")
figure.show()

# What is the relationships between the sales price of iPhone and the number of rating on Flipkart?

In [ ]:
data.head(2)

Product Name  \
0         APPLE iPhone 8 Plus (Gold, 64 GB)   
1  APPLE iPhone 8 Plus (Space Grey, 256 GB)   

                                         Product URL  Brand  Sale Price  \
0  https://www.flipkart.com/apple-iphone-8-plus-g...  Apple       49900   
1  https://www.flipkart.com/apple-iphone-8-plus-s...  Apple       84900   

     Mrp  Discount Percentage  Number Of Ratings  Number Of Reviews  \
0  49900                    0               3431                356   
1  84900                    0               3431                356   

                Upc  Star Rating   Ram  
0  MOBEXRGV7EHHTGUH          4.6  2 GB  
1  MOBEXRGVAC6TJT4F          4.6  2 GB

In [ ]:
figure = px.scatter(data_frame =data, x ="Number Of Ratings",y = "Sale Price",size='Discount Percentage',trendline='ols',title='Relationship Between Sale Price and Number of Ratings')
figure.show()

There is a negative linear relationship between the sales price of iPhone and the number of ratings. It means iPhones with lower sales prices are sold more in india.

# What is the relationships between the Discount percentage of iPhone and the number of rating on Flipkart?

In [ ]:
figure = px.scatter(data_frame = data, x = "Number Of Ratings",y = "Discount Percentage", size = "Sale Price", trendline = "ols", title = "Relationship Between Discount Percentage and Number of Ratings")
figure.show()

# Most Expensive and least expensive

In [ ]:
most_expensive = data.loc[data['Sale Price'].idxmax()]
least_expensive = data.loc[data['Sale Price'].idxmin()]

#Display the results
print("Most Expensive iPhone:")
print(most_expensive)

print("\nLeast Expensive iPhone:")
print(least_expensive)


Most Expensive iPhone:
Product Name                        APPLE iPhone 12 Pro (Silver, 512 GB)
Product URL            https://www.flipkart.com/apple-iphone-12-pro-s...
Brand                                                              Apple
Sale Price                                                        140900
Mrp                                                               149900
Discount Percentage                                                    6
Number Of Ratings                                                    542
Number Of Reviews                                                     42
Upc                                                     MOBFWBYZ5UY6ZBVA
Star Rating                                                          4.5
Ram                                                                 4 GB
Name: 24, dtype: object

Least Expensive iPhone:
Product Name                              APPLE iPhone SE (White, 64 GB)
Product URL            https://www.flipkart.com/appl